presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestProgressMonitor.java

```diff
@@ -0,0 +1,172 @@
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestProgressMonitor
+{
+    private static final String SERVER_ADDRESS = "127.0.0.1:8080";
+
+    private static final String QUERY_ID = "20160128_214710_00012_rk68b";
+    private static final String INFO_URI = "http://" + SERVER_ADDRESS + "/v1/query/" + QUERY_ID;
+    private static final String PARTIAL_CANCEL_URI = "http://" + SERVER_ADDRESS + "/v1/stage/" + QUERY_ID + ".%d";
+    private static final String NEXT_URI = "http://" + SERVER_ADDRESS + "/v1/statement/" + QUERY_ID + "/%d";
+    private static final List<Column> RESPONSE_COLUMNS = ImmutableList.of(new Column("_col0", "bigint", new ClientTypeSignature("bigint", ImmutableList.of())));
+    private static final List<String> RESPONSES = ImmutableList.of(
+            newQueryResults(null, 1, null, null, "QUEUED"),
+            newQueryResults(1, 2, RESPONSE_COLUMNS, null, "RUNNING"),
+            newQueryResults(1, 3, RESPONSE_COLUMNS, null, "RUNNING"),
+            newQueryResults(0, 4, RESPONSE_COLUMNS, ImmutableList.of(ImmutableList.of(253161)), "RUNNING"),
+            newQueryResults(null, null, RESPONSE_COLUMNS, null, "FINISHED"));
+
+    private static String newQueryResults(Integer partialCancelId, Integer nextUriId, List<Column> responseColumns, List<List<Object>> data, String state)
+    {
+        try {
+            QueryResults queryResults = new QueryResults(
+                    QUERY_ID,
+                    new URI(INFO_URI),
+                    partialCancelId == null ? null : new URI(String.format(PARTIAL_CANCEL_URI, partialCancelId)),
+                    nextUriId == null ? null : new URI(String.format(NEXT_URI, nextUriId)),
+                    responseColumns,
+                    data,
+                    new StatementStats(state, true, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, null),
+                    null,
+                    null,
+                    null);
+
+            ObjectMapper mapper = new ObjectMapper();
+            StringWriter stringWriter = new StringWriter();
+            mapper.writeValue(stringWriter, queryResults);
+            return stringWriter.toString();
+        }
+        catch (Exception e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    @Test
+    public void test()
+            throws Exception
+    {
+        try (Connection connection = createConnection()) {
+            try (Statement statement = connection.createStatement()) {
+                PrestoStatement prestoStatement = statement.unwrap(PrestoStatement.class);
+                RecordingProgressMonitor progressMonitor = new RecordingProgressMonitor();
+                prestoStatement.setProgressMonitor(progressMonitor);
+                try (ResultSet rs = statement.executeQuery("SELECT a bogus query")) {
```

Comment:
Maybe the string `"bogus query for testing"` so it's even more obvious (no `SELECT`)